lib/realtime_web/channels/realtime_channel/broadcast_handler.ex

```diff
@@ -23,24 +24,26 @@ defmodule RealtimeWeb.RealtimeChannel.BroadcastHandler do
         self_broadcast: self_broadcast,
         tenant_topic: tenant_topic,
         authorization_context: authorization_context,
-        db_conn: db_conn
+        tenant: tenant_id
       }
     } = socket
 
-    case run_authorization_check(socket, db_conn, authorization_context) do
-      {:ok,
-       %{assigns: %{ack_broadcast: ack_broadcast, policies: %Policies{broadcast: %BroadcastPolicies{write: true}}}} =
-           socket} ->
-        socket = increment_rate_counter(socket)
-        send_message(self_broadcast, tenant_topic, payload)
-        if ack_broadcast, do: {:reply, :ok, socket}, else: {:noreply, socket}
+    with {:ok, db_conn} <- Connect.lookup_or_start_connection(tenant_id) do
```

Comment:
This with without an else will return `{:error, _reason}` but the `@spec` says that it must return `{:reply` or `{:noreply`. 

I was expecting dialyzer to complain there 🤔